src/libponyrt/lang/stdfd.c

```diff
@@ -512,7 +512,7 @@ PONY_API void pony_os_std_print(FILE* fp, char* buffer, size_t len)
   if(len == 0)
     return;
 
-  fprintf(fp, "%s\n", buffer);
+  fprintf(fp, "%*.*s\n", (int)len, (int)len, buffer);
```

Comment:
I'd expect you to only need `%*s\n` here, does that not work?

```c
fprintf(fp, "%*s\n", (int)len, buffer);
```

Reply:
This formatting string creates a problem for certain Strings. See also https://ponylang.zulipchat.com/#narrow/stream/189985-beginner-help/topic/Bug.20in.20String.2Efrom_array.28.29.20.3F/near/286809387